### DISCUSSION TITLE
Storybook HMR issues with recipe import + clean: true

### DISCUSSION THREAD

**Discussion by adbutterfield** [chakra-ui/panda#2484] [Q&A] [answered]
### Description

I have a monorepo with tokens and recipes in one package, and react components in another package (maybe another package with vue components using the same recipes soon).

The HMR is working, but there's a small bug. When I have `clean: true` in the `panda.config.ts` of my react package, when I change a recipe in the tokens package, the `styled-system` directory gets wiped out, and gets replaced with only the one recipe that I updated in the tokens package.

I guess HMR is trying to be smart, and only update the files that need to be updated, but since clean runs, we actually need to regenerate everything.

I guess we need some way to disable `clean` in development? 

### Link to Reproduction

Sorry, no reproduction...

### Panda CSS Version

0.37.1

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

**Comment by segunadebayo** [comment, marked as answer]
What’s the motivation to add clean by default? Would recommend only adding clean when needed, say,  when you run codegen.

Alternatively, like you said, turn off clean based on an env variable.

**Reply by adbutterfield** [reply]
I guess you only really need it before codegen or build. I'm also using tsup which has a `clean` option, but doesn't have the same behavior. 
I guess maybe they do a full rebuild and not HMR when you pass `--watch`?

Anyways, this might not be a real problem, if it makes sense to have the current behavior.
